homeassistant/components/lcn/config_flow.py

```diff
@@ -106,4 +134,76 @@ async def async_step_import(self, data: ConfigType) -> ConfigFlowResult:
             self.hass.config_entries.async_update_entry(entry, data=data)
             return self.async_abort(reason="existing_configuration_updated")
 
-        return self.async_create_entry(title=f"{host_name}", data=data)
+        return self.async_create_entry(title=f"{data[CONF_HOST]}", data=data)
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> config_entries.ConfigFlowResult:
+        """Handle a flow initiated by the user."""
+        if user_input is None:
+            return self.async_show_form(step_id="user", data_schema=USER_SCHEMA)
+
+        errors = None
+        if get_config_entry(self.hass, user_input):
+            errors = {CONF_BASE: "already_configured"}
+        elif (error := await validate_connection(user_input)) is not None:
+            errors = {CONF_BASE: error}
+
+        if errors is not None:
+            return self.async_show_form(
+                step_id="user",
+                data_schema=self.add_suggested_values_to_schema(
+                    USER_SCHEMA, user_input
+                ),
+                errors=errors,
+            )
+
+        data: dict = {
+            **user_input,
+            CONF_DEVICES: [],
+            CONF_ENTITIES: [],
+        }
+
+        return self.async_create_entry(title=data[CONF_HOST], data=data)
+
+
+class LcnOptionsFlowHandler(config_entries.OptionsFlow):
+    """Handle LCN options."""
+
+    def __init__(self, config_entry: config_entries.ConfigEntry) -> None:
+        """Initialize LCN options flow."""
+        self.config_entry = config_entry
+
+    async def async_step_init(
+        self, user_input: dict[str, Any] | None = None
+    ) -> config_entries.ConfigFlowResult:
+        """Manage the LCN options."""
+        errors = None
+        if user_input is not None:
+            user_input[CONF_HOST] = self.config_entry.data[CONF_HOST]
+
+            await self.hass.config_entries.async_unload(self.config_entry.entry_id)
+            if (error := await validate_connection(user_input)) is not None:
+                errors = {CONF_BASE: error}
+
+            # brief delay to allow host free up used license for validation
+            await asyncio.sleep(0.5)
```

Comment:
This looks like a race, can't we check if what we're waiting for has happened instead?

Reply:
That's indeed a relict from testing. I removed this. I guess it's not necessary as it never occured during my tests.